cpp/CMakeLists.txt

```diff
@@ -213,6 +221,32 @@ add_library(cudf SHARED
 #Override RPATH for cudf
 SET_TARGET_PROPERTIES(cudf PROPERTIES BUILD_RPATH "\$ORIGIN")
 
+###################################################################################################
+# - jitify ----------------------------------------------------------------------------------------
+
+# Creates executable stringify and uses it to convert types.h to c-str for use in JIT code
+add_executable(stringify "${CMAKE_SOURCE_DIR}/thirdparty/jitify/stringify.cpp")
+execute_process(WORKING_DIRECTORY ${CMAKE_BINARY_DIR}
+    COMMAND ${CMAKE_COMMAND} -E make_directory ${CMAKE_BINARY_DIR}/include)
+
+add_custom_command(OUTPUT ${CMAKE_BINARY_DIR}/include/types.h.jit
+                   WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/include
+                   COMMAND ${CMAKE_BINARY_DIR}/stringify cudf/types.h > ${CMAKE_BINARY_DIR}/include/types.h.jit
+                   COMMENT "stringify header types.h"
+                   DEPENDS stringify
+                   MAIN_DEPENDENCY ${CMAKE_CURRENT_SOURCE_DIR}/include/cudf/types.h)
+
+add_custom_target(stringify_run DEPENDS ${CMAKE_BINARY_DIR}/include/types.h.jit)
```

Comment:
`DEPENDS` does not have to be a path, it can be a CMake variable.

For example:
```
add_custom_command(OUTPUT  TYPES_JIT
	                   WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/include
	                   COMMAND ${CMAKE_BINARY_DIR}/stringify cudf/types.h > ${CMAKE_BINARY_DIR}/include/types.h.jit
                           ...)
...
add_custom_target(stringify_run DEPENDS TYPES_JIT)
...
```

As long as the `MAIN_DEPENDENCY` is properly specified to be `${CMAKE_CURRENT_SOURCE_DIR}/include/cudf/types.h`, CMake will update `jit_types` if that file changes.

Note that this is helpful if later you expect your command to have multiple outputs.

I would also consider renaming `stringify_run` to something more descriptive. I'm not really sure what the purpose of this target is off the cuff, and that obfuscates our build process for new developers, etc.

Reply:
I tried this but it didn't work. Running `make` rebuilds `launcher.cpp` every time.
```cmake
add_custom_command(OUTPUT STRINGIFIED_HEADERS
                   WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/include
                   COMMAND ${CMAKE_BINARY_DIR}/stringify cudf/types.h > ${CMAKE_BINARY_DIR}/include/types.h.jit
                   COMMENT "Run stringify on header types.h to convert it to c-str for use in JIT compiled code"
                   DEPENDS stringify
                   MAIN_DEPENDENCY ${CMAKE_CURRENT_SOURCE_DIR}/include/cudf/types.h)

add_custom_target(stringified_headers DEPENDS STRINGIFIED_HEADERS)

add_dependencies(cudf stringified_headers)

```

Reply:
Sorry what does the cmake code above have to do with launcher.cpp?

Reply:
This code is for making and running an executable that comes with Jitify. This executable, called stringify, converts source files into c-strings and writes them to another source file. Now those stringified source files can be used in JIT code compilation. In our specific use case, we wanted to be able to use the definitions in `types.h` in the JIT kernels. So we use stringify to convert `types.h` into a string and store it in `types.h.jit` in the build/include directory. That `types.h.jit` is included in `launcher.cpp`. I wanted a way to use CMake to run stringify only when `types.h` is changed. With @mt-jones ' suggestion, CMake runs it all the time and every time i run `make`, `launcher.cpp` is rebuilt because the include is touched.